db/Pos/DB/Functions.hs

```diff
@@ -41,6 +43,23 @@ dbGetBi tag key = do
 dbPutBi :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> m ()
 dbPutBi tag k v = dbPut tag k (serialize' v)
 
+-- | Read serialized value (with version) associated with given key from pure DB.
+dbGetBiWithVersion
+    :: forall v m.
+       (Bi v, MonadDBRead m)
+    => DBTag -> ByteString -> m (Maybe (Word8, v))
+dbGetBiWithVersion tag key = do
+    bytes <- dbGet tag key
+    traverse (dbDecode . (ToDecodeValue key)) bytes
+
+-- | Write serializable value to DB for given key. Uses simple versioning.
+dbPutBiWithVersion :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> Word8 -> m ()
+dbPutBiWithVersion tag k v version = dbPut tag k (dbSerialize version v)
+
+-- | Version of 'serialize'' function that includes version when serializing a value.
+dbSerialize :: Bi a => Word8 -> a -> ByteString
+dbSerialize version = serialize' . (version,)
```

Comment:
Why does this function takes `Word8` argument if we know that for this software it must always be 0? I think this 0 should be in `constants.yaml`.